core/src/main/java/org/opensearch/sql/expression/function/BuiltinFunctionRepository.java

```diff
@@ -64,25 +67,45 @@ public class BuiltinFunctionRepository {
    *
    * @return singleton instance
    */
-  public static synchronized BuiltinFunctionRepository getInstance() {
-    if (instance == null) {
-      instance = new BuiltinFunctionRepository(new HashMap<>());
+  public static synchronized BuiltinFunctionRepository getInstance(
+      DataSourceService dataSourceService) {
+    Set<DataSourceMetadata> dataSourceMetadataSet = dataSourceService.getDataSourceMetadata(true);
+    Set<Integer> dataSourceServiceHashSet =
+        dataSourceMetadataSet.stream()
+            .map(metadata -> metadata.hashCode())
```

Comment:
I think it could be optimized to
```suggestion
            .map(Object::hashCode)
```